From 5c5f7596dd088ba88999a2bdaed22a5af013c289 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 10 Nov 2018 16:18:39 +0100
Subject: [PATCH 1/1] efi_loader: create fdt reservation before copy

When copying the device we must ensure that the copy does not fall into a
memory area reserved by the same.

So let's change the sequence: first create memory reservations and then
copy the device tree.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 cmd/bootefi.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/cmd/bootefi.c b/cmd/bootefi.c
index 8673e742b58..51b542ac54a 100644
--- a/cmd/bootefi.c
+++ b/cmd/bootefi.c
@@ -312,6 +312,9 @@ static efi_status_t efi_install_fdt(ulong fdt_addr)
 		return EFI_INVALID_PARAMETER;
 	}
 
+	/* Create memory reservation as indicated by the device tree */
+	efi_carve_out_dt_rsv(fdt);
+
 	/* Prepare fdt for payload */
 	ret = copy_fdt(&fdt);
 	if (ret)
@@ -322,8 +325,6 @@ static efi_status_t efi_install_fdt(ulong fdt_addr)
 		return EFI_LOAD_ERROR;
 	}
 
-	efi_carve_out_dt_rsv(fdt);
-
 	/* Link to it in the efi tables */
 	ret = efi_install_configuration_table(&efi_guid_fdt, fdt);
 	if (ret != EFI_SUCCESS)
-- 
2.19.1

